source/client-side-encryption/tests/README.rst

```diff
@@ -1080,3 +1161,143 @@ Invalid Hostname in KMS Certificate
    In Python, this message is "certificate verify failed: IP address mismatch, certificate is not valid for '127.0.0.1'". In Go, this message
    is "cannot validate certificate for 127.0.0.1 because it doesn't contain any IP SANs". If the language of implementation has a single, generic
    error message for all certificate validation errors, drivers may inspect other fields of the error to verify its meaning.
+
+KMS TLS Options Test
+~~~~~~~~~~~~~~~~~~~~
+
+Two mock KMS server processes must be running:
+
+1. The mock `KMS HTTP server <https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/csfle/kms_http_server.py>`_.
+
+   Run on port 8002 with `ca.pem`_ as a CA file and `server.pem`_ as a cert file.
+
+   Run with the ``--require_client_cert`` option.
+   
+   Example:
+
+   .. code::
+
+      python -u kms_http_server.py --ca_file ../x509gen/ca.pem --cert_file ../x509gen/server.pem --port 8002 --require_client_cert
+   
+
+2. The mock `KMS KMIP server <https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/csfle/kms_kmip_server.py>`_.
+
+
+Create a ``ClientEncryption`` object (referred to as ``client_encryption``)
```

Comment:
A caveat of these `KMS TLS Options Test` is that they do not test TLS options configured Automatic Encryption.

Tests for Automatic Encryption are more roundabout, but may be feasible. I think it would require manually inserting a document with a chosen endpoint, like:

```
 {
  "status":  1,
  "_id": (...),
  "masterKey": {
    "region": "us-east-1",
    "key": "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
    "provider": "aws",
    "endpoint": "127.0.0.1:8002" // Endpoint for test.
  },
  "updateDate": (...),
  "keyMaterial": (...),
  "creationDate": (...)
}

```

Then attempt to insert with automatic encryption.

I chose to hold off on this. C [shares a code path](https://github.com/kevinAlbs/mongo-c-driver/blob/74717c2a1477be2011b49d73fd5c88bb17289aa0/src/libmongoc/src/mongoc/mongoc-crypt.c#L835:L865) for configuring TLS options through ClientEncryption and AutomaticEncryption.

I am happy to add tests for Automatic Encryption if they add value and if this method of testing is OK.

Reply:
I think it's fine this way.